TILE_AUTHOR_GUIDE.md

```diff
@@ -0,0 +1,598 @@
+// this type is not the real one in the source.
+type PathTemplateData struct{
+    // Name is set to the BOSH Release name
+    Name string
+    
+    // Version is set to the BOSH Release version
+    Version string
+
+    // Name is set to the Kilnfile.lock StemcellCriteria OS value
+    StemcellOS string
+
+    // Name is set to the Kilnfile.lock StemcellCriteria Version value
+    StemcellVersion string
+}
+```
+
+Here are some example path templates (the rest of the release source config has been omitted).
+
+```yaml
+release_sources:
+  - path_template: "bla-bla/{{.Name}}/{{.Version}}/{{.Name}}-{{.Version}}-{{.StemcellOS}}-{{.StemcellVersion}}.tgz"
+  - path_template: "bla-bla/{{.Name}}/{{.Version}}/{{.Name}}-{{.Version}}.tgz"
+  - path_template: "bla-bla/{{.Name}}-{{.Version}}-{{.StemcellOS}}-{{.StemcellVersion}}.tgz"
+```
+
+Avoid using other Go text/template actions like `{{if pipeline}}` and the like.
+
+#### <a id="release-source-boshio"></a> BOSH.io
+
+Kiln can only download releases from BOSH.io and can not upload BOSH Releases to BOSH.io.
+
+This release source has minimal configuration.
+Just add it to your `release_sources` and you can get releases from [BOSH.io](https://bosh.io/releases/).
+
+```yaml
+# Expected Kilnfile.lock
+release_sources:
+  - type: bosh.io
+    id: community # (optional) the default ID for this type is the constant string "bosh.io"
+```
+
+The value of `remote_path` in the BOSH release tarball lock is a URL.
+
+
+#### <a id='release-source-github'></a> GitHub Release Artifacts
+
+Kiln can only download releases from GitHub Releases and can not upload BOSH Releases to BOSH.io.
+
+To download BOSH Release Tarballs from GitHub Releases, add the following
+```
+release_sources:
+  - type: "github"
+    id: crhntr # (optional) the default ID in this case is the value of org
+    org: "crhntr"
+    github_token: $(variable "github_token")
+```
+
+You will need one entry per organization.
+Some examples are: "pivotal", "cloudfoundry", "pivotal-cf", or your personal/company GitHub username.
+
+You will need to add the following flag to most commands:
+
+```
+# Optional helper
+export GITHUB_TOKEN="$(gh auth status --show-token 2>&1 | grep 'Token:' | awk '{print $NF}')"
+
+# Example Kiln variable flag
+kiln fetch --variable="github_token=${GITHUB_TOKEN}"
+```
+
+The value of `remote_path` in the BOSH release tarball lock is a URL.
+
+#### <a id='release-source-artifactory'></a> Build Artifactory
+
+Kiln can fetch and upload releases to/from Build Artifactory.
+
+The release source specification should look like this:
+
+```yaml
+release_sources:
+  - type: "artifactory"
+    id: "official-storage"  # (optional) the default ID for this type is the constant string value "artifactory"
+    repo: "some-repository"
+    username: "some-username"
+    password: "some-password"
+    path_template: "some-path-template/tarball.tgz"
+```
+
+Note `repo` is not a GitHub repository. It is an Artifactory thing. 
```

Comment:
I would vouch for renaming `It is an Artifactory thing` to `It refers to an Artifactory repository`

Reply:
Thanks :)